config/confighttp/confighttp.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package confighttp
+
+import (
+	"net/http"
+	"time"
+
+	"go.opentelemetry.io/collector/config/configtls"
+)
+
+type HTTPClientSettings struct {
+	// The target to which the exporter is going to send traces or metrics,
+	// using the gRPC protocol. The valid syntax is described at
+	// https://github.com/grpc/grpc/blob/master/doc/naming.md.
+	Endpoint string `mapstructure:"endpoint"`
+
+	// TLSSetting struct exposes TLS client configuration.
+	TLSSetting configtls.TLSClientSetting `mapstructure:",squash"`
+
+	// Timeout parameter configures `http.Client.Timeout`.
+	Timeout time.Duration `mapstructure:"timeout,omitempty"`
+}
+
+func (hcs *HTTPClientSettings) ToClient() (*http.Client, error) {
+	tlsCfg, err := hcs.TLSSetting.LoadTLSConfig()
```

Comment:
When I last discussed this with @ccaraman the idea was that whether to enable TLS for clients would depend on the user specifying it in the `endpoint`. e.g. with some config like this:

```
...
endpoint: https://localhost:9090
timeout: 5s
```

Do you agree this is how we should distinguish between turning on TLS or not? If so this would need to change to check the scheme of `endpoint`.

Reply:
I think user may want to use Unix sockets or other transport and the enabling of the TLS should not depend on the address used